src/main/groovy/com/bmuschko/gradle/docker/utils/DockerThreadContextClassLoader.groovy

```diff
@@ -59,6 +58,22 @@ class DockerThreadContextClassLoader implements ThreadContextClassLoader {
     }
 
     /**
+     * Checks if Docker host URL starts with http(s) and if so, converts it to tcp
+     * which is accepted by docker-java library.
+     *
+     * @param dockerClientConfiguration docker client configuration
+     * @return Docker host URL as string
+     */
+    private String getDockerHostUrl(DockerClientConfiguration dockerClientConfiguration) {
+        String dockerUrl = dockerClientConfiguration.url ?: dockerExtension.url
+        if(dockerUrl.toLowerCase().startsWith("http")) {
+            dockerUrl = "tcp" + dockerUrl.substring(dockerUrl.indexOf(":"))
+        }
+
+        dockerUrl
```

Comment:
Lets use the ternary operator and shorten this up a bit:

    String url = (dockerClientConfiguration.url ?: dockerExtension.url).toLowerCase()
    url.startsWith("http") ? "tcp" + url.substring(url.indexOf(":")) : url